### PR TITLE
difficulty filters

### PR DESCRIPTION
**Pull request by rogerbutt**

### CODE REVIEW

File: client/src/containers/AllCode.js [hunk 1]
```diff
@@ -19,28 +20,48 @@ class AllCode extends Component {
...
+
+    componentWillReceiveProps(nextProps) {
+        let { dispatch, codes } = nextProps;
+    }
+
+    changeDifficulty(diff) {
+        this.setState({
+            difficulty: diff,
+        });
+        console.log(diff);
```
Comment (rong2) [inline client/src/containers/AllCode.js]: Can you remove console.log?

File: client/src/reducers/code.js [hunk 2]
```diff
@@ -47,6 +49,18 @@ export default function codes(state = initialState, action) {
                 isFetchingByUserId: true
             });
 
+        case FETCH_ALL_CODE:
+            return Object.assign({}, state, {
+                isFetching: true,
+            });
+
+        case RECEIVE_ALL_CODE:
+            console.log(action);
```
Comment (rong2) [inline client/src/reducers/code.js]: same as above?

File: client/src/containers/AllCode.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import React, { PropTypes, Component } from 'react'
-import request from 'superagent';
```
Comment (rong2) [inline client/src/containers/AllCode.js:2]: Was this used for anything?

Comment (rogerbutt) [inline client/src/containers/AllCode.js:2]: No so it was removed?

File: client/src/containers/AllCode.js [hunk 4]
```diff
@@ -49,4 +69,12 @@ class AllCode extends Component {
     };
 }
 
-export default connect()(AllCode)
+function mapPropsToState(state) {
+    let codes = state.code.codes || [];
+
+    return {
+        codes
```
Comment (rong2) [inline client/src/containers/AllCode.js:80]: Do you not need to define a PropType for codes?

Comment (rogerbutt) [inline client/src/containers/AllCode.js:80]: you're right

**Pull request merged**